backend/enhanced_stock_retrieval_working.py

```
@@ -0,0 +1,1557 @@
+    try:
+        import django  # type: ignore
+        os.environ.setdefault('DJANGO_SETTINGS_MODULE', os.environ.get('DJANGO_SETTINGS_MODULE', 'stockscanner_django.settings'))
+        django.setup()
+        from django.utils import timezone as tz  # type: ignore
+        from stocks.models import Stock as StockModel, StockPrice as StockPriceModel  # type: ignore
+        django_timezone = tz
+        Stock = StockModel
+        StockPrice = StockPriceModel
+        DJANGO_READY = True
+        logger.info("Django initialized for DB operations")
+    except Exception as e:  # pragma: no cover
```

Ruff (0.14.3)

Do not catch blind exception: `Exception`

(BLE001)